overlord/snapstate/snapstate.go

```diff
@@ -1218,13 +1218,16 @@ func InstallMany(st *state.State, names []string, userID int, flags *Flags) ([]s
 		return nil, nil, err
 	}
 
+	var transLane int
+	if flags.Transactional {
+		transLane = st.NewLane()
+	}
 	tasksets := make([]*state.TaskSet, 0, len(installs))
 	for _, sar := range installs {
 		info := sar.Info
 		var snapst SnapState
-		var flags Flags
 
-		flags, err := ensureInstallPreconditions(st, info, flags, &snapst)
+		flags, err := ensureInstallPreconditions(st, info, *flags, &snapst)
```

Comment:
this change doesn't look quite right, as the result will be the input to the next ensureInstallPreconditions, instead the input should always be the original value. We might need a test to prove the difference

Reply:
Oh, right, I overlooked this was in a per-snap loop. Changed now.